src/ec/eddsa.rs

```diff
@@ -18,9 +18,76 @@
 
 use {bssl, c, signature, signature_impl};
 use input::Input;
+use rand::{RAND, SecureRandom};
 
 struct EdDSA;
 
+/// An Ed25519 key pair.
+//
+// This can be used for signing. Allows generation of a new key pair through
+// the `generate()` method, or using an existing key pair with `from_bytes()`.
+pub struct Ed25519KeyPair {
+    private_public: [u8; 64],
+}
+
+
+impl<'a> Ed25519KeyPair {
+    /// Generate a new key pair from randomness.
+    pub fn generate(rng: &SecureRandom) -> Ed25519KeyPair {
+        let mut rand = RAND::new(rng);
+        let mut pair = Ed25519KeyPair { private_public: [0; 64] };
+        let mut public_key = [0u8; 32];
+        unsafe {
+            ED25519_keypair(public_key.as_mut_ptr(),
+                            pair.private_public.as_mut_ptr(),
+                            &mut rand);
+        };
+        debug_assert_eq!(pair.private_public[32..], public_key[..]);
+        pair
+    }
+
+    /// Copies key data from the given slices to create a new key pair.
+    pub fn from_bytes(private_key: &[u8; 32], public_key: &[u8; 32]) -> Ed25519KeyPair {
+        let mut pair = Ed25519KeyPair { private_public: [0; 64] };
+        {
```

Comment:
I suggest you use `let (private_key, public_key) = pair.private_public.split_at(32)` to simplify this function.